### DISCUSSION TITLE
Unable to access MariaDB

### DISCUSSION THREAD

**Discussion by goloroden** [community/community#25546] [Actions]
<p>Hey there :wave:</p>
<p>I have a really strange problem, and I’m stuck now after having tried to solve this for hours.</p>
<p>The problem: I have a GitHub Actions-based build, which needs to start a MariaDB and connect to it. I start the MariaDB with the following command:</p>
<pre><code>docker run -d -p 3309:3306 -e MYSQL_ROOT_PASSWORD=wolkenkit -e MYSQL_USER=wolkenkit -e MYSQL_PASSWORD=wolkenkit -e MYSQL_DATABASE=wolkenkit --name test-mariadb mariadb:10.4.10 --bind-address=0.0.0.0
</code></pre>
<p>This works fine. Afterwards I try to connect (after a sleep of 15 seconds to give the container enough time to start):</p>
<pre><code>mysql --port=3009 --user=wolkenkit --password=wolkenkit --database=wolkenkit --verbose --execute="SELECT USER(),CURRENT_USER();"
</code></pre>
<p>This works locally on my macOS machine. But it does not work on GitHub Actions, instead I get an error message telling me:</p>
<pre><code>ERROR 1045 (28000): Access denied for user 'wolkenkit'@'localhost' (using password: YES)
##[error]Process completed with exit code 1.
</code></pre>
<p>Why is that? What am I missing?</p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Please try to add – <strong>protocol=tcp</strong> setting to mysql command. Because you are running MySQL inside Docker container, socket is not available and you need to connect through TCP. Setting “–protocol” in the mysql command will change that.</p>
<p>And you specify a wrong port 3009 in mysql command, please correct it to 3309.  </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/2/2c63b9966fb7edcaf8e067fb6153a87963bd6e35.png" data-download-href="/uploads/short-url/6kGFrq1qyjsBhUE47gPQClsWvZz.png?dl=1" title="mysql.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181091778-b42ce39b-2eb3-484b-9d28-aac05f5d677b.png" alt="mysql.png" data-base62-sha1="6kGFrq1qyjsBhUE47gPQClsWvZz" width="690" height="305" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/2/2c63b9966fb7edcaf8e067fb6153a87963bd6e35_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">mysql.png</span><span class="informations">1217×538 11.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by goloroden** [comment]
<p>Oh… yes, the port is wrong :flushed:</p>
<p>It’s *these* things that keep you trying for hours, and of course you look everywhere but there :wink:</p>
<p>Fixing the port, setting the protocol and increasing the timeout helped.</p>
<p>Thanks :+1:</p>
